src/parse.py

```diff
@@ -40,6 +40,25 @@
     # Regardless of the above case, here's how the file name should terminate
     '(\s|$|:)+'
 )))
+MASTER_REGEX_WITH_SPACES = re.compile(''.join((
+    #begin the capture
+    '(',
+    #a leading / for absolute dirs if its there
+    '\/?',
+    # now we look at directories, where we allow either normal chars
+    # or a single whitespace char followed by legit chars
+    '(([a-z.A-Z0-9\-_]|\s[a-zA-Z0-9\-_])+\/)+',
```

Comment:
This is good, but unfortunately real directories can have multiple spaces. I'm not dinging the approach - the regex otherwise would overmatch on everything - and clearly it's an improvement. But '/just for you pcottle/' is a valid directory.  Something to think about.